tests/core/engine_adapter/test_redshift.py

```diff
@@ -29,6 +29,32 @@ def test_columns(adapter: t.Callable):
     assert resp == {"col": exp.DataType.build("INT")}
 
 
+def test_varchar_size_workaround(make_mocked_engine_adapter: t.Callable, mocker: MockerFixture):
```

Comment:
Can we also have a test that validates this flow all the way to the table creation expression?